orchagent/watermarkorch.cpp

```diff
@@ -107,41 +107,50 @@ void WatermarkOrch::doTask(NotificationConsumer &consumer)
 
     consumer.pop(op, data, values);
 
-    if(&consumer == m_clearNotificationConsumer){
-        string wm_name = "";
-        vector<sai_object_id_t> &obj_ids = m_pg_ids;
```

Comment:
The root cause of the bug is the use of reference, and rebind it (obj_ids = m_pg_ids; obj_ids = m_unicast_queue_ids;)?

Reply:
Yes, the clearSingleWm() was called with wrong vector of oids.